packages/web-ui/src/sections/DocumentsSidebar/useTree/index.ts

```diff
@@ -0,0 +1,118 @@
+import { useMemo } from 'react'
+
+export type SidebarDocument = {
+  path: string
+  doumentUuid: string
+}
+
+export class Node {
+  public id: string
+  public name: string
+  public isRoot: boolean = false
+  public children: Node[] = []
+  public doc?: SidebarDocument
+
+  constructor({
+    id,
+    doc,
+    children = [],
+    isRoot = false,
+    name = '',
+  }: {
+    id: string
```

Comment:
what is this id? do we need it?

Reply:
for `key` react elements. Is nice to do `key={node.id}`